app/permission/foreground/main.go

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: Unlicense OR MIT
+
+/*
+Package foreground implements permissions to run a foreground service.
+See https://developer.android.com/guide/components/foreground-services.
+
+The following entries will be added to AndroidManifest.xml:
+
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
+
+*/
+
+package foreground
+
+// Start notifies the system that the program will perform
+// background work and that it shouldn't be killed. It returns a channel
+// that should be closed when the background work is complete.
+
+// Start is a no-op on Linux, Windows, macOS; Android will
+// display a notification during background work; iOS isn't supported.
+func Start(id int, title, text string) chan struct{} {
```

Comment:
Why let the app specify the channel ID, which is Android specific? I think hard-coding it is OK.

Reply:
the id gives you a reference to the notification in case of wanting to modify/update it late. yes this is android specific.